src/java/org/apache/cassandra/cql3/QueryProcessor.java

```diff
@@ -989,10 +989,10 @@ else if (statement instanceof SelectStatement)
                 statementKsName = selectStatement.keyspace();
                 statementCfName = selectStatement.table();
             }
-            else if (statement instanceof BatchStatement)
+            else if (statement instanceof CompositeCQLStatement)
             {
-                BatchStatement batchStatement = ((BatchStatement) statement);
-                for (ModificationStatement stmt : batchStatement.getStatements())
+                CompositeCQLStatement transactionStatement = (CompositeCQLStatement) statement;
+                for (CQLStatement stmt : transactionStatement.getStatements())
```

Comment:
nit: `transactionStatement` -> `batchStatement` ?

Reply:
removed the variable, so no worries on names anymore =D